src/topp/DatabaseSuitability.cpp

```diff
@@ -149,55 +152,46 @@ class DatabaseSuitability :
 
     for (const auto& pep_id : pep_ids)
     {
-      vector<PeptideHit> hits = pep_id.getHits();
+      const vector<PeptideHit>& hits = pep_id.getHits();
 
       if (hits.empty()) continue;
 
-      PeptideHit top_hit = hits[0];
+      const PeptideHit& top_hit = hits[0];
 
+      // skip if the top hit is a decoy hit
       if (top_hit.getMetaValue("target_decoy") == "decoy") continue;
 
       // check if top hit is found in de novo protein
-      set<String> accessions = top_hit.extractProteinAccessionsSet();
-      bool is_novo = true;
-      for (const String& acc : accessions)
+      if (!isNovoHit_(top_hit)) // top hit is db hit
       {
-        if (acc.find(Constants::UserParam::CONCAT_PEPTIDE) == String::npos)
-        {
-          is_novo = false;
-          break;
-        }
+        ++count_db;
+        continue;
       }
-
-      if (is_novo) // top hit is de novo hit
+      else // top hit is novo hit
       {
         if (hits.size() == 1)
         {
           ++count_novo;
           continue;
         }
 
-        PeptideHit second_hit = hits[1];
-
-        if (second_hit.getMetaValue("target_decoy") == "decoy")
+        // find the second target hit, skip all decoy hits inbetween
+        const PeptideHit* second_hit = nullptr;
+        for (Int i = 1; i < hits.size(); ++i)
+        {
+          if (hits[i].getMetaValue("target_decoy") == "target")
```

Comment:
Hm as I mentioned. Here it probably would be better to do something like:
startsWith("target") in case you have a "target+decoy" peptide.

Reply:
Yes, this will come in the next commit! :)